sda-mq/templates/tests.yaml

```diff
@@ -0,0 +1,84 @@
+{{- define "mqfullname" -}}
+{{- if .Values.fullnameOverride -}}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name:  "{{ .Release.Name }}-test-certs"
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+type: Opaque
+data:
+  {{- (.Files.Glob "files/{tester,CA,ca}*").AsSecrets | nindent 2 }}
+
+---
+
+apiVersion: v1
+kind: Pod
```

Comment:
shouldn't this be a `job` better than a pod as we can get completed status ?
Or I am missing something.

Similar comment for other tests.

Reply:
I'm not sure I understand this correctly. Maybe we're not all synchronized on what is desired. So far I think all resources should be cleaned up (at least on successful test) and thus there is no saved job/pod status, but rather it's displayed when the tests complete:

```
$ helm test postgres
NAME: postgres
LAST DEPLOYED: Fri Oct 16 15:20:14 2020
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE:     postgres-test-certs
Last Started:   Fri Oct 16 16:58:51 2020
Last Completed: Fri Oct 16 16:58:51 2020
Phase:          Succeeded
TEST SUITE:     postgres-test
Last Started:   Fri Oct 16 16:58:51 2020
Last Completed: Fri Oct 16 16:58:57 2020
Phase:          Succeeded
$ 
```

is something more persistent desired?

Reply:
no, I thought helm tests make use of `job` from k8s but apparently they use `pod` so I think there was not enough info on my side.

Reply:
Ah. Actually, it can use both - and I suspect it can use any "runnable" resource. I'm not sure how it keeps track of the difference between resources like secrets or configmaps and things like jobs and pods (that run and possibly give an output status), but a useful way of thinking about it seems to be that he resources annotated with test will be created on helm test (and cleaned up afterwards according to annotations).